homeassistant/components/fritzbox/coordinator.py

```diff
@@ -95,6 +116,13 @@ def _update_fritz_devices(self) -> FritzboxCoordinatorData:
         self.new_devices = device_data.keys() - self.data.devices.keys()
         self.new_templates = template_data.keys() - self.data.templates.keys()
 
+        if (
+            (not self.data.devices and not self.data.templates)
```

Comment:
What is the point of this? self.data is the old stuff right. Why remove devices lf we dont have any?

Reply:
this is to ensure, even after a restart of the integration, that we check if there are any obsolete devices in the registry left

Reply:
With the above change i think you should always run the cleanup.

Reply:
the condition `(not self.data.devices and not self.data.templates)` will only match on first coordinator data fetch, because there are no previous data, but we already collected the current/new data, so we are able to check the device registry for orphan entries ... on every upcoming coordinator data fetch, we only check for orphan device registry entries, if there are no more existing `devices` or `templates`.

Reply:
Ok. I would probably just always do a check on first refresh inside async setup instead of coordninator. But okey.

Reply:
mhhhh 🤔  sounds reasonable ... will add a async setup method to the coordinator 👍

Reply:
I think there was some proposals for a init method for coordinator. Called only on first refresh. Not sure if that got merged yet.

Reply:
b4decb18b3da2b4f26d4cf189be0341166747e77